parachains/common/src/xcm_config.rs

```diff
@@ -65,3 +70,32 @@ impl ShouldExecute for DenyReserveTransferToRelayChain {
 		Ok(())
 	}
 }
+
+/// A `ChargeFeeInFungibles` implementation that converts the output of
+/// a given WeightToFee implementation an amount charged in
+/// a particular assetId from pallet-assets
+pub struct AssetFeeAsExistentialDepositMultiplier<R, WeightToFee, CON>(
+	PhantomData<(R, WeightToFee, CON)>,
+);
+impl<CurrencyBalance, R, WeightToFee, CON>
+	cumulus_primitives_utility::ChargeWeightInFungibles<AccountIdOf<R>, pallet_assets::Pallet<R>>
+	for AssetFeeAsExistentialDepositMultiplier<R, WeightToFee, CON>
+where
+	R: pallet_assets::Config,
+	WeightToFee: WeightToFeePolynomial<Balance = CurrencyBalance>,
+	CON: BalanceConversion<
+		CurrencyBalance,
+		<pallet_assets::Pallet<R> as Inspect<AccountIdOf<R>>>::AssetId,
+		<pallet_assets::Pallet<R> as Inspect<AccountIdOf<R>>>::Balance,
+	>,
+	AccountIdOf<R>:
+		From<polkadot_primitives::v2::AccountId> + Into<polkadot_primitives::v2::AccountId>,
+{
+	fn charge_weight_in_fungibles(
+		asset_id: <pallet_assets::Pallet<R> as Inspect<AccountIdOf<R>>>::AssetId,
+		weight: Weight,
+	) -> Result<<pallet_assets::Pallet<R> as Inspect<AccountIdOf<R>>>::Balance, XcmError> {
+		let amount = WeightToFee::calc(&weight);
+		CON::to_asset_balance(amount, asset_id).map_err(|_| XcmError::TooExpensive)
```

Comment:
If we are going to use `BalanceToAssetBalance::to_asset_balance` I think we would have to review the `minimum_balance` of the sufficient Assets we currently have in Statemine/t.

It can lead to very unbalanced fees if the `min_balance` ratios are not well adjusted to the assets price, encouraging people to use a specific asset over others to pay the fees.

**E.g**:
- KSM 
  - existential deposit = 3,333,333
  - decimals = 12  
- USDT
  - minimum balance = 1,000
  - decimals = 6
  
A fee of 1,000,000,000,000 (1 KSM) is equivalent to  1,000,000,000,000 * 1,000/3,333,333 = 300,000,000 USDT (300 USDT)

However, 1 KSM currently is 73$. It means that paying fees with KSM would be x5 times cheaper .

Please let know if the numbers are correct.

Reply:
This is already a problem with the existing pallet_asset_tx_payment 
 https://github.com/paritytech/cumulus/blob/8befd605b6a29210cb47c74ed484e90f90ebaaac/parachains/runtimes/assets/statemine/src/lib.rs#L501 no?

In that sense, this PR does not introduce any change. However, my personal opinion is that we should review definitely the minimum amount of sufficient assets. The numbers seem definitely correct to me.